Realm/RLMQueryUtil.mm

```diff
@@ -1850,6 +1848,46 @@ bool is_self_value_for_key_path_function_expression(NSExpression *expression)
                        @"Only support compound, comparison, and constant predicates");
     }
 }
+
+// This function returns the nested subscripts from a NSPredicate with the following format `anyCol[0]['key'][#any]`
+// and its respective keypath (including any linked keypath)
+// This will iterate each argument of the NSExpression and its nested NSExpressions, takes the constant subscript
+// and creates a PathElement to be used in the query. If we use `#any` as a wildcard this will show in the parser 
+// predicate as NSKeyPathExpressionType.
+NSString* QueryBuilder::get_path_elements(std::vector<PathElement> &paths, NSExpression *expression) {
+    NSString *keyPath = @"";
+    for (NSUInteger i = 0; i < expression.arguments.count; i++) {
+        NSString *nestedKeyPath = @"";
+        if (expression.arguments[i].expressionType == NSFunctionExpressionType) {
+            nestedKeyPath = get_path_elements(paths, expression.arguments[i]);
+        } else if (expression.arguments[i].expressionType == NSConstantValueExpressionType) {
+            id value = [expression.arguments[i] constantValue];
+            if ([value isKindOfClass:[NSNumber class]]) {
+                paths.push_back(PathElement{[(NSNumber *)value intValue]});
+            } else if ([value isKindOfClass:[NSString class]]) {
+                NSString *key = (NSString *)value;
+                if ([key isEqual:@"#any"]) {
```

Comment:
This is not the correct thing to be checking for. This is matching `['#any']` rather than `[#any]`. The whole point here is to be using something that *isn't* a string value which could be a legal dictionary key. The tests for this are also all incorrect. Creating this via substitution syntax requires using `%K`, not `%@`.